src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/Miscellaneous/AutoLayoutStructureTests.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class AutoLayoutStructureTests
+    {
+        [StructLayout(LayoutKind.Auto)]
+        public struct SomeTestStruct_Auto
+        {
+            public int i;
+        }
+
+        [STAThread]
+        [Fact]
+        public void AutoLayoutStructureTest()
+        {
+            SomeTestStruct_Auto someTs_Auto = new SomeTestStruct_Auto();
+            try
+            {
+                Marshal.StructureToPtr(someTs_Auto, new IntPtr(123), true);
+                Assert.True(false, "Marshal.StructureToPtr did not throw an exception.");
+            }
+            catch (ArgumentException ex)
+            {
+                Assert.True(ex.ParamName == "structure", "Thrown ArgumentException is incorrect.");
+                Assert.True(ex.Message.Contains("The specified structure must be blittable or have layout information."), "Thrown ArgumentException is incorrect.");
+            }
+            catch (Exception)
+            {
+                Assert.True(false, "Marshal.StructureToPtr threw unexpected exception.");
+            }
```

Comment:
This whole test would be better structured using AssertExtensions.Throws and Assert.Contains, e.g.
```C#
var someTs_Auto = new SomeTestStruct_Auto();
ArgumentException ex = AssertExtensions.Throws<ArgumentException>("structure", () => Marshal.StructureToPtr(someTs_Auto, new IntPtr(123), true));
Assert.Contains("The specified structure must be blittable or have layout information."), ex.Message);
```
(That said, for that last assert, what happens if the exception message is localized?)

Reply:
@stephentoub How about simply looking looking for "blittable" - I don't think that is normally translated?

Reply:
Some cultures tend to translate **everything**, including words like blittable. E.g. https://docs.microsoft.com/ru-ru/dotnet/framework/interop/copying-and-pinning#formatted-blittable-classes

Reply:
I've re-structured the code, but still need to consider about the exception message